### ISSUE TITLE
Amazon Braket backend support

### ISSUE THREAD

**Issue by JordanAWS** [mit-han-lab/torchquantum#137]
Adding in an plugin to support Amazon Braket like the qiskit ones here: https://github.com/mit-han-lab/torchquantum/tree/main/torchquantum/plugins

**Comment by Hanrui-Wang** [comment]
Thank you very much for the help. I have assigned @AbdullahKazi500

**Comment by JordanAWS** [comment]
Thanks @AbdullahKazi500! Please don't hesitate to reach out if you have questions :)

**Comment by kvathupo** [comment]
@AbdullahKazi500 Ah, my sincerest apologies. Bonne chance !

**Comment by Amana-L** [comment]
Hi @JordanAWS, I would like to work on this. Are you looking to commit the Amazon Braket plugin to the linked repository i.e. torchquantum?

**Comment by JordanAWS** [comment]
Since multiple people (@AbdullahKazi500 and @Amana-L ) have volunteered for this issue, I will assign it to each of you.

Here are the rules: you can either
A) Work together on this issue and split the bounty 
B) Each work independently and race to the finish line. The first person to submit a working PR (including tests) will win the full bounty.

Please let me know what you each prefer by reacting to to this message with an 🅰️ or 🅱️ :)

If a subset of hackers react A, I will take that to mean they are forming a team to work on this issue, and if anyone else reacts B, I will take that to mean that hacker is working independently.

**Comment by JordanAWS** [comment]
And yes, the idea here is to submit a Braket plugin in the pytorch repo.

**Comment by JordanAWS** [comment]
@AbdullahKazi500 and @Amana-L -- do you need any support to help complete this issue?

**Comment by Amana-L** [comment]
Hi @JordanAWS, thanks for checking up. 

I'm currently working on it, but I don't know if I can get it done before the UnitaryHack deadline. I'll submit it anyway though :)

Is there a link to the Braket documentation page? I can't seem to find it. Its quite a lot of operations and it would be nice to have all the classes and their methods in one place.

**Comment by JordanAWS** [comment]
https://docs.aws.amazon.com/braket/latest/developerguide/what-is-braket.html Here you go @Amana-L !

**Comment by JordanAWS** [comment]
@Amana-L It looks like you weren't quite able to get a PR in for this issue before the June 13th deadline for UnitaryHACK, but if you're still interested in working on this issue, we can offer you AWS credits as an alternative to the bounty :)

**Comment by Amana-L** [comment]
@JordanAWS Yes, I wasn't able to get through in time. I'm still working on it though and the AWS credits are much appreciated!

**Comment by nathanshammah** [comment]
@Amana-L if the PRs are accepted in the next few days, the bounties will be awarded.

**Comment by Hanrui-Wang** [comment]
Dear all, 

Thank you for the valuable discussions! I just want to friendly remind that this issue is actually not one of the five issues we posted on the UnitaryHack website. But since this is such an important functionality, I will assign a $50 bounty by myself for a minimum worked version of Braket backend, with more bounty for advanced functions.

**Comment by JordanAWS** [comment]
@Hanrui-Wang It is one of the bounties posted as part of the Amazon Braket project actually :) https://unitaryhack.dev/projects/amazon-braket-sdk/

@Amana-L If you'd like credits, you'll need to create an AWS account (if you don't have one already), and share your account ID with unitary-hack-braket-support@amazon.com with a request for credits to work on the issue :)
